menas/src/main/scala/za/co/absa/enceladus/menas/repositories/OozieRepository.scala

```diff
@@ -197,12 +198,13 @@ class OozieRepository @Autowired() (oozieClientRes: Either[OozieConfigurationExc
         }
 
         val in = connection.getInputStream
-        val targetArray = Array.fill(in.available)(0.toByte)
-        in.read(targetArray)
         val os = hadoopFS.create(hadoopPath, true)
-        os.write(targetArray)
+
+        IOUtils.copy(in, os)
+
         os.flush()
         os.close()
+        in.close()
```

Comment:
Isn't it a good pattern to have the close after IO in `finally` block. To ensure closing even if something happens. 🤔